### PR TITLE
Add option to link statically with hwloc on linux

### PR DESCRIPTION
**Pull request by igchor**
### Checklist

- [ ] Code compiles without errors locally
- [ ] All tests pass locally
- [ ] CI workflows execute properly

- [ ] CI workflows, not executed per PR (e.g. Nightly), execute properly 
- [ ] New tests added, especially if they will fail without my changes
- [ ] Added/extended example(s) to cover this functionality
- [ ] Extended the README/documentation
- [ ] All newly added source files have a license
- [ ] All newly added source files are referenced in CMake files
- [ ] Logger (with debug/info/... messages) is used

### CODE REVIEW

**Review by PatKamin — APPROVED** [review]

File: compile_hwloc.sh [hunk 1]
```diff
@@ -0,0 +1,2 @@
+CFLAGS="-fPIC" make
```
Comment (lplewa) [inline compile_hwloc.sh:1]: Move it to scripts dir? or Cmake.

**Comment by KFilipek** [comment]
@igchor
Following change allows me to compile your case without any issue:
```
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0c3a5e8..cf86d13 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -146,7 +146,7 @@ else()
     add_custom_command(
         COMMAND
             ./configure --prefix=${hwloc_targ_BINARY_DIR} --enable-static=yes
-            --enable-shared=no --disable-libxml2 --disable-levelzero
+            --enable-shared=no --disable-libxml2 --disable-levelzero CFLAGS=-fPIC CXXFLAGS=-fPIC
         WORKING_DIRECTORY ${hwloc_targ_SOURCE_DIR}
         OUTPUT ${hwloc_targ_SOURCE_DIR}/Makefile
         DEPENDS ${hwloc_targ_SOURCE_DIR}/configure)
```

File: compile_hwloc.sh (hunk 1 above)
Comment (igchor) [reply compile_hwloc.sh]: Removed, all is done in CMake now

**Comment by igchor** [comment]
> @igchor Following change allows me to compile your case without any issue:
> 
> ```
> diff --git a/CMakeLists.txt b/CMakeLists.txt
> index 0c3a5e8..cf86d13 100644
> --- a/CMakeLists.txt
> +++ b/CMakeLists.txt
> @@ -146,7 +146,7 @@ else()
>      add_custom_command(
>          COMMAND
>              ./configure --prefix=${hwloc_targ_BINARY_DIR} --enable-static=yes
> -            --enable-shared=no --disable-libxml2 --disable-levelzero
> +            --enable-shared=no --disable-libxml2 --disable-levelzero CFLAGS=-fPIC CXXFLAGS=-fPIC
>          WORKING_DIRECTORY ${hwloc_targ_SOURCE_DIR}
>          OUTPUT ${hwloc_targ_SOURCE_DIR}/Makefile
>          DEPENDS ${hwloc_targ_SOURCE_DIR}/configure)
> ```

Thanks, applied.

File: CMakeLists.txt [hunk 2]
```diff
@@ -110,10 +110,11 @@ if(NOT UMF_LINK_HWLOC_STATICALLY)
...
         )
     endif()
-else()
-    if(NOT WINDOWS)
-        message(FATAL_ERROR "hwloc can be statically linked only on Windows")
-    endif()
+    # add PATH to DLL on Windows
+    set(DLL_PATH_LIST
+        "${DLL_PATH_LIST};PATH=path_list_append:${LIBHWLOC_LIBRARY_DIRS}/../bin"
+    )
```
Comment (lukaszstolarczuk) [inline CMakeLists.txt:116]: if this flag works on non-Windows OSes, please update the `UMF_LINK_HWLOC_STATICALLY` description in CMake and in README

**Review by bratpiorka — APPROVED** [review]

File: CMakeLists.txt [hunk 3]
```diff
@@ -137,6 +138,46 @@ else()
     message(STATUS "    LIBHWLOC_LIBRARIES = ${LIBHWLOC_LIBRARIES}")
```
Comment (lukaszstolarczuk) [inline CMakeLists.txt:138]: these messages could be moved below the if, or copy-pasted, or something 😉

File: CMakeLists.txt (hunk 2 above)
Comment (igchor) [reply CMakeLists.txt:118]: Done.

File: CMakeLists.txt (hunk 3 above)
Comment (igchor) [reply CMakeLists.txt:140]: Done.

Comment (lukaszstolarczuk) [reply CMakeLists.txt:140]: thx!

File: test/CMakeLists.txt [hunk 4]
```diff
@@ -271,7 +271,8 @@ add_umf_test(
 # tests for the proxy library
 if(UMF_PROXY_LIB_ENABLED
    AND UMF_BUILD_SHARED_LIBRARY
-   AND NOT UMF_DISABLE_HWLOC)
+   AND NOT UMF_DISABLE_HWLOC
+   AND NOT UMF_LINK_HWLOC_STATICALLY)
```
Comment (vinser52) [inline test/CMakeLists.txt:275]: Why do we need to check that `NOT UMF_LINK_HWLOC_STATICALLY`?

**Review by vinser52 — APPROVED** [review]

File: CMakeLists.txt [hunk 5]
```diff
@@ -110,10 +112,11 @@ if(NOT UMF_LINK_HWLOC_STATICALLY)
             "${DLL_PATH_LIST};PATH=path_list_append:${LIBHWLOC_LIBRARY_DIRS}/../bin"
         )
     endif()
-else()
-    if(NOT WINDOWS)
-        message(FATAL_ERROR "hwloc can be statically linked only on Windows")
-    endif()
+    # add PATH to DLL on Windows
+    set(DLL_PATH_LIST
```
Comment (lukaszstolarczuk) [inline CMakeLists.txt:116]: nit: this seems like a redundant code...?

**Review by lukaszstolarczuk — COMMENTED** [review]
LGTM 👍

File: test/CMakeLists.txt (hunk 4 above)
Comment (bratpiorka) [reply test/CMakeLists.txt:275]: @vinser52 - as I recall there were some non-trivial issues with building proxy lib with static hwloc. I think we should address this later

**Review by bratpiorka — APPROVED** [review]

**Pull request merged**